modules/gdnative/godot/godot_rid.cpp

```diff
@@ -28,36 +28,36 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 #include "godot_rid.h"
+#include "core/variant.h"
 
-#include "object.h"
-#include "resource.h"
+#include "core/rid.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-void _rid_api_anchor() {
-}
-
-void GDAPI godot_rid_new(godot_rid *p_rid, godot_object *p_from) {
+void _rid_api_anchor() {}
 
-	Resource *res_from = ((Object *)p_from)->cast_to<Resource>();
-
-	RID *rid = (RID *)p_rid;
-	memnew_placement(rid, RID);
+void GDAPI godot_rid_new(godot_rid *r_dest) {
+	RID *dest = (RID *)r_dest;
+	memnew_placement(dest, RID);
+}
```

Comment:
The old function wasn't useless, take a look at http://docs.godotengine.org/en/stable/classes/class_rid.html.

With this API there's no way to get the RID of an object.

We could have *both* functions, but *only* this is bad.